File: server/handlers/points/tokenchecker.go
// server/api/auth.go
package points

import (
	"encoding/json"
	"net/http"
)

func AccessChecker(client string) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Access-Control-Allow-Origin", client)
		w.Header().Set("Access-Control-Allow-Credentials", "true")

		_, err := r.Cookie("access_token")
		if err != nil {
			http.Error(w, "access token не найден", http.StatusUnauthorized)
			return
		}

		response := map[string]bool{
			"authenticated": true,
		}

		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(response)
	}
}

File: client/src/modules/tokens/tokenchecker.js
async function checktoken() {
  try {
    let response = await fetch("http://localhost:8080/accesschecker", {
      credentials: "include",
    });

    if (response.ok) {
      const data = await response.json();
      return data.authenticated === true;
    }

    const refreshResponse = await fetch("http://localhost:8080/refresh", {
      credentials: "include",
    });

    if (refreshResponse.ok) {
      const secondResponse = await fetch(
        "http://localhost:8080/accesschecker",
        {
          credentials: "include",
        }
      );

      if (secondResponse.ok) {
        const data = await secondResponse.json();
        return data.authenticated === true;
      }
    }

    return false;
  } catch (error) {
    console.error("Token check error:", error);
    return false;
  }
}

export { checktoken };
